PROTOCOL-690: Feature/aave claim aave

dumebi opened the pull request:
Implemented `AaveClaimAaveFacet` to claim accrued aave coins

dumebi committed:
chore: aave claimAave solidity code

dumebi committed:
fix test on polygon

dumebi committed:
fixed review comments

passabilities commented on config/dapps.ts:
```diff
@@ -2,16 +2,18 @@ import { Tokens } from '../types/custom/config-types'
   sushiswapV2RouterAddress: '0x1b02dA8Cb0d097eB8D57A175b88c7D8b47997506',
   uniswapV2RouterAddress: '0x0000000000000000000000000000000000000000',
-  compoundComptrollerAddress: '',
+  compoundComptrollerAddress: '0x3d9819210a31b4961b30ef54be2aed79b9c9cd3b',
```
@dumebi pretty sure Compound is not on Polygon?

dumebi replied:
fixed

passabilities commented on test/helpers/story/drivers/dapp-utils/aave.utils.ts:
```diff
@@ -57,6 +62,34 @@ async function withdrawAave(
+  const aaveAfter = await IncentiveController.getUserUnclaimedRewards(
+    escrowAddress
+  )
+  expect(aaveAfter.toString()).to.equal('0')
```
be sure to check that the claimed balance went to the user's address

dumebi replied:
done

dumebi replied:
done

dumebi committed:
chore: redeem staked aave

dumebi committed:
remove compound address from polygon

sypeer commented on contracts/escrow/dapps/AaveClaimAaveFacet.sol:
```diff
@@ -0,0 +1,103 @@
+    );
+
+    /**
+     * @notice To claim comp call the claim comp function on COMPTROLLER_ADDRESS_PROVIDER_ADDRESS.
```
should be aave in place of comp

dumebi replied:
fixed

dumebi committed:
remove documentation relative to comp on claimAave

dumebi committed:
chore: add getNetworkName to protocol

dumebi committed:
rebased feature/create-loan-nft-v2

dumebi committed:
rebased with feature/create-loan-nft-v2

passabilities committed:
Fix chai BN array assertion

passabilities committed:
Add Aave incentives controller for Polygon

passabilities committed:
Update BN equal check

passabilities committed:
Fix comment

passabilities committed:
Update Compound & Aave claim

passabilities committed:
Remove file

dumebi committed:
claim aave and comp conditionally to user or escrow

passabilities commented on contracts/escrow/dapps/CompoundClaimComp.sol:
```diff
@@ -48,7 +48,12 @@ contract CompoundClaimCompFacet is PausableMods, DappMods {
+
+        // Add AAVE to escrow token list
+        // TODO: get the COMP token (add as immutable constructor variable)
+        LibEscrow.tokenUpdated(loanID, address(0));
```
this should be throwing an error in the tests when called.. but its not.

this function calls `ERC20(tokenAddress).balanceOf()` which should revert since `address(0)` is not a contract.

can we verify that the tests are working @dumebi

dumebi replied:
This is because we only claim comp (to pass) when we have accrued comp. and during the test, the escrow does not have accrued comp

passabilities replied:
we should then also have a test that advances time so that it can accrue COMP

dumebi replied:
fixing these on `feature/claimAaveCompToUser`

dumebi committed:
patch fix accrued comp and aave

admazzola commented on contracts/escrow/dapps/AaveClaimAaveFacet.sol:
```diff
@@ -0,0 +1,102 @@
+        public
+        view
+        paused("", false)
+        onlyBorrower(loanID)
```
is onlyBorrower necessary for a View method ?

dumebi replied:
fixing these on `feature/claimAaveCompToUser`

dumebi committed:
fix: comp  and aave to user

dumebi committed:
aave lending pool

dumebi committed:
fix review changes

dumebi committed:
fix `tokenUpdated` parameters

dumebi committed:
fixed parameter reviews

updated `tokenUpdated` function to include comp
added function params for comp and aave

dumebi committed:
remove console.log

dumebi committed:
testing compAccrued
